packages/dashboard/src/components/eventDetailsSidebar.tsx

```diff
@@ -0,0 +1,63 @@
+// EventDetailsSidebar.tsx
+import React from "react";
+import { Drawer, Typography, Box } from "@mui/material";
+type SelectedEvent = {
+  [x: string]: any;
+  messageId: string;
+  eventType: string;
+  event: string;
+  userId: string | null;
+  anonymousId: string | null;
+  processingTime: string;
+  eventTime: string;
+  traits: string;
+};
+interface EventDetailsSidebarProps {
+  open: boolean;
+  onClose: () => void;
+  selectedEvent: SelectedEvent | null;
+}
+
+const EventDetailsSidebar: React.FC<EventDetailsSidebarProps> = ({
+  open,
+  onClose,
+  selectedEvent,
+}) => {
+  return (
+    <Drawer
+      open={open}
+      onClose={onClose}
+      anchor="right"
+      slotProps={{ backdrop: { onClick: onClose } }}
+    >
+      <Box padding={2} paddingTop={10}>
+        <Typography fontFamily={"monospace"} variant="h2">
+          Event Details
+        </Typography>
+        {selectedEvent &&
+          Object.keys(selectedEvent).map((key) => {
+            return key !== "traits" ? (
+              <Typography key={key} fontFamily={"monospace"}>
+                {`${key}: ${selectedEvent[key as keyof SelectedEvent]}`}
+              </Typography>
+            ) : (
+              <></>
+            );
+          })}
+
+        {selectedEvent && selectedEvent.traits && (
+          <>
+            <Typography fontFamily={"monospace"}>Traits:</Typography>
```

Comment:
Can you use react code mirror, as demonstrated in `packages/dashboard/src/components/templateEditor.tsx` to render this json?

Reply:
As discussed in chat

> the purpose of codemirror is that it just provides a better presentation. You can use codemirror in a view only mode, so that the contained code (json in this case) is static and not editable.